src/main/java/appeng/util/item/ItemList.java

```diff
@@ -158,99 +117,83 @@ public boolean isEmpty()
 		return !this.iterator().hasNext();
 	}
 
-	public Collection<StackType> findFuzzyDamage( AEItemStack filter, FuzzyMode fuzzy, boolean ignoreMeta )
-	{
-		StackType low = (StackType) filter.getLow( fuzzy, ignoreMeta );
-		StackType high = (StackType) filter.getHigh( fuzzy, ignoreMeta );
-		return this.records.subMap( low, true, high, true ).descendingMap().values();
-	}
-
 	@Override
-	public synchronized void addStorage( StackType option ) // adds a stack as
-	// stored.
+	public void addStorage( IAEItemStack option )
 	{
-		if( this.checkStackType( option ) )
+		if( this.isStackNull( option ) )
 		{
 			return;
 		}
 
-		StackType st = this.records.get( option );
+		final IAEItemStack st = this.getItemRecord( option.getItem() ).get( option );
 
 		if( st != null )
 		{
-			// st.setPriority( currentPriority );
 			st.incStackSize( option.getStackSize() );
 			return;
 		}
 
-		StackType opt = (StackType) option.copy();
-		// opt.setPriority( currentPriority );
-		this.records.put( opt, opt );
+		final IAEItemStack opt = (IAEItemStack) option.copy();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unnecessary cast to "IAEItemStack". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)